pkg/ottl/parser.go

```diff
@@ -56,12 +56,40 @@ func (s *Statement[K]) Execute(ctx context.Context, tCtx K) (any, bool, error) {
 	return result, condition, nil
 }
 
-func NewParser[K any](functions map[string]interface{}, pathParser PathExpressionParser[K], enumParser EnumParser, telemetrySettings component.TelemetrySettings) Parser[K] {
+func NewParser[K any](
+	functions map[string]interface{},
+	pathParser PathExpressionParser[K],
+	options ...Option,
+) Parser[K] {
+	var cfg config
+	for _, opt := range options {
+		opt(&cfg)
+	}
+
 	return Parser[K]{
 		functions:         functions,
 		pathParser:        pathParser,
-		enumParser:        enumParser,
-		telemetrySettings: telemetrySettings,
+		enumParser:        cfg.enumParser,
+		telemetrySettings: cfg.telemetrySettings,
+	}
+}
+
+type config struct {
```

Comment:
What is the behavior when any of these are not present? I think we should throw an error when parsing statements that require an option that wasn't passed.